### DISCUSSION TITLE
Pi Zero W as the Linux server?

### DISCUSSION THREAD

**Discussion by garethmarlow** [tesla-android/issue-tracker#20] [Q&A]
Hi - does a Pi Zero W have enough power to act as the Linux server? I have Tesla Android up and running on a Pi 4, but I currently only have a Pi Zero W kicking around, and I'd prefer not to have to buy another Pi 3 if the Linux server would only be needed while the project's in Alpha.

Great work btw - excited already to see CarPlay running on the Android Pi.

**Comment by garethmarlow** [comment]
![image](https://user-images.githubusercontent.com/59932701/168399093-da8906e2-738d-44ee-9a1b-b77de74fd147.jpeg)

I'm getting console errors when I try to boot the Pi Zero from the flashed SD card image. "Stopped killswitch manager" is repeated several times before the screen blanks and the Pi becomes unresponsive.

**Comment by mikegapinski** [comment]
it should work on Pi Zero W - but not out of the box. Network interfaces are different etc. 

I would suggest starting with a clean raspbian image and going step by step with adjustments if needed. Tesla-Android-OS image generation is done here:

https://github.com/tesla-android/tesla-android-os/blob/main/src/modules/tesla-android/start_chroot_script

I guess that all should work but you won't need the bridge interface(br0) config in /etc/network/interfaces and you Android Pi will be connecting to the Wi-Fi network instead of using ethernet.

If we figure out the steps needed for Zero W I'll cook an image for it

**Reply by garethmarlow** [reply]
OK I've borrowed a Pi 4 from my son to get it all working, so I'll go back to the Pi W over the next few days and try to get it working from the raspbian base. 

How long do you expect the second Pi to be needed for?

**Reply by mikegapinski** [reply]
A few months, I'll be updating the current dual board version and migrating components one by one. 

I'll make myself a deadline when I'll have most of the stuff done

**Comment by garethmarlow** [comment]
Confirmed: Raspberry Pi Zero W (v1) works as the Tesla Android OS (linux/raspbian) server. For some reason, pihole-FTL won't start if eth0 isn't present (works if you plug in a USB ethernet dongle), but will work if you use pihole -up to upgrade it on the image.

Steps to get it working:
1. Follow existing steps from the install guide, up to step 12 (sudo brctl addif br0 eth0) - this line is not required
2. In your terminal, type "pihole -up". Pihole will upgrade itself, pulling new packages down over your LTE 4G connection.
3. Once this has completed successfully, type "sudo reboot" at the terminal
4. Reconnect the Android Pi to monitor and mouse. Start this machine
5. Connect the Android Pi to the Linux Pi's wireless network using the Android Setting menu
6. Connect your computer to the Linux Pi's wireless network and access pihole via http://3.3.3.1/admin/
7. Find the DHCP leases page. You should see the Android Pi's wireless connection in the "Currently active DHCP leases" table.
8. Click the icon to the right of the "trash" icon, to transfer the MAC address and other details to the Static leases table further down in the page
9. For this new entry, change the IP address to 3.3.3.50 and the hostname to Android. Click "Save"
10. Disconnect and reconnect the Android Pi to the wireless network. Confirm that it has received the "3.3.3.50" IP address
11. Power down both devices, connect them to your car and off you go

Only issue so far is that I can only access lower quality mode via the browser, regardless of whether the page was refreshed while the car was in park.
